ShellPkg/Library/UefiShellCommandLib/UefiShellCommandLib.c

```diff
@@ -398,15 +398,16 @@ ShellCommandIsCommandOnInternalList (
         ; Node = (SHELL_COMMAND_INTERNAL_LIST_ENTRY *)GetNextNode (&mCommandList.Link, &Node->Link)
         )
   {
-    ASSERT (Node->CommandString != NULL);
-    if (gUnicodeCollation->StriColl (
-                             gUnicodeCollation,
-                             (CHAR16 *)CommandString,
-                             Node->CommandString
-                             ) == 0
-        )
-    {
-      return (TRUE);
+    if (Node->CommandString != NULL) {

```

Comment:
IIUC, in DEBUG builds, this assumes having a Shell command without any string associated with it is valid as we just check the next command . On the other hand I agree that RELEASE code would just crash.

Maybe this would allow to do both ?
```
    if (Node->CommandString == NULL) {
      ASSERT (FALSE);
      continue;
    }

    if (gUnicodeCollation->StriColl (
                              gUnicodeCollation,
                              (CHAR16 *)CommandString,
                              Node->CommandString
                              ) == 0
        )
    {
      return (TRUE);
    }
```
(comment applies for all the changes where an ASSERT was removed)